core/types/src/macros.rs

```diff
@@ -68,6 +68,83 @@ macro_rules! impl_newtype {
     )*}
 }
 
+/// This macro provides common byte-handling operations when the type being
+/// wrapped is a struct containing a single fixed-size array of bytes.
+///
+/// This should be called from within a private submodule.
+/// TODO: Delete this once Display has been implemented for all callers.
```

Comment:
:wrench: I understand the desire is that this is meant to be a temp placeholder, but temp placeholders have a habit of persisting at times.
Consider updating `impl_newtype_for_bytestruct` to be something like:
```rust
#[macro_export]
macro_rules! impl_newtype_for_bytestruct {
    ($($wrapper:ident, $inner:ident, $size:ident, $fieldname:ident;)*) => {$(

        $crate::impl_newtype_for_bytestruct_no_display! {
            $wrapper, $inner, $size, $fieldname;
        }

        impl ::core::fmt::Display for $wrapper {
            fn fmt(&self, f: &mut ::core::fmt::Formatter) -> ::core::fmt::Result {
                write!(f, "{:#X}", self)
            }
        }

    )*}
}
```